packages/forms/components/TextInput.tsx

```diff
@@ -0,0 +1,125 @@
+import * as React from "react";
+import { cx } from "emotion";
+import { inputAppearances, inputContainer, inputValidation } from "../style";
+import {
+  display,
+  flex,
+  flexItem,
+  flush,
+  inputReset,
+  margin,
+  padding,
+  textWeight
+} from "../../shared/styles/styleUtils";
+
+export enum TextInputAppearance {
+  Standard = "standard",
+  Error = "error",
+  Success = "success"
+}
+
+export interface TextInputProps extends React.HTMLProps<HTMLInputElement> {
+  id: string;
+  type: "text" | "number" | "search" | "email" | "password" | "tel" | "url";
+  appearance: TextInputAppearance;
+  inputLabel: string | React.ReactNode;
+  showInputLabel: boolean;
+  validationContent?: string | React.ReactNode;
+}
+
+export class TextInput<T extends TextInputProps> extends React.PureComponent<
+  T,
+  {}
+> {
+  public static defaultProps: Partial<TextInputProps> = {
+    type: "text",
+    appearance: TextInputAppearance.Standard,
+    showInputLabel: true
+  };
+
+  public render() {
+    const labelContent = this.getLabelContent();
+    const containerProps: { className?: string } = {};
+    if (this.props.className) {
+      containerProps.className = this.props.className;
+    }
+    return (
+      <div {...containerProps}>
+        {labelContent}
+        {this.getInputContent()}
+        {this.getValidationContent()}
+      </div>
+    );
+  }
+
+  protected getInputAppearance() {
+    return this.props.disabled ? "disabled" : this.props.appearance;
+  }
+
+  protected getLabelContent() {
+    return (
+      <label
+        className={cx(
+          flush("top"),
+          margin("bottom", "xxs"),
+          textWeight("medium"),
+          display(this.props.showInputLabel ? "block" : "none")
```

Comment:
`display: none` and `visibility: hidden` will cause assistive technology to ignore the label (e.g.: causes screen readers not to read the label aloud)

We should add something to style utils to handle this. Maybe something like...

packages/shared/styles/styleUtils/modifiers/visuallyHidden.ts:
````
export const visuallyHidden = css`
		border: 0;
		clip: rect(0 0 0 0);
		position: absolute;
		overflow: hidden;
		margin: -1px;
		padding: 0;
		width: 1px;
		height: 1px;
`
````